server/app/controllers/admin/AdminProgramStatusesController.java

```diff
@@ -4,28 +4,51 @@
 import static com.google.common.base.Preconditions.checkNotNull;
 
 import auth.Authorizers;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
 import com.google.inject.Inject;
 import controllers.CiviFormController;
+import forms.admin.ProgramStatusesEditForm;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+import org.apache.commons.lang3.tuple.Pair;
 import org.pac4j.play.java.Secure;
+import play.data.DynamicForm;
+import play.data.Form;
+import play.data.FormFactory;
 import play.mvc.Http;
+import play.mvc.Http.HttpVerbs;
 import play.mvc.Result;
+import services.CiviFormError;
+import services.ErrorAnd;
+import services.LocalizedStrings;
+import services.program.ProgramDefinition;
 import services.program.ProgramNotFoundException;
 import services.program.ProgramService;
+import services.program.StatusDefinitions;
 import views.admin.programs.ProgramStatusesView;
 
 public final class AdminProgramStatusesController extends CiviFormController {
```

Comment:
While I'm putting the comment on the class, I think the controller needs to document its API better.  EG nothing declares what Adding a status does (eg append), or replace (maintains position, and changes the admin visible text / non-translated content)